src/Symfony/Component/Notifier/Bridge/KazInfoTeh/Tests/KazInfoTehTransportTest.php

```diff
@@ -0,0 +1,98 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Notifier\Bridge\KazInfoTeh\Tests;
+
+use Symfony\Component\HttpClient\MockHttpClient;
+use Symfony\Component\HttpClient\Response\MockResponse;
+use Symfony\Component\Notifier\Bridge\KazInfoTeh\KazInfoTehTransport;
+use Symfony\Component\Notifier\Exception\TransportException;
+use Symfony\Component\Notifier\Message\MessageInterface;
+use Symfony\Component\Notifier\Message\SmsMessage;
+use Symfony\Component\Notifier\Test\TransportTestCase;
+use Symfony\Component\Notifier\Transport\TransportInterface;
+use Symfony\Contracts\HttpClient\HttpClientInterface;
+
+/**
+ * @author Egor Taranov <dev@taranovegor.com>
+ */
+final class KazInfoTehTransportTest extends TransportTestCase
+{
+    public function createTransport(HttpClientInterface $client = null): TransportInterface
+    {
+        return (new KazInfoTehTransport(
+            'username',
+            'password',
+            'sender',
+            $client ?? $this->createMock(HttpClientInterface::class)
+        ))->setHost('test.host');
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function toStringProvider(): iterable
+    {
+        yield ['kazinfoteh://test.host?sender=sender', $this->createTransport()];
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function supportedMessagesProvider(): iterable
+    {
+        yield [new SmsMessage('77000000000', 'KazInfoTeh!')];
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function unsupportedMessagesProvider(): iterable
+    {
+        yield [new SmsMessage('420000000000', 'KazInfoTeh!')];
+
+        yield [$this->createMock(MessageInterface::class)];
+    }
+
+    public function createClient(int $statusCode, string $content): HttpClientInterface
+    {
+        return new MockHttpClient(new MockResponse($content, ['http_code' => $statusCode]));
+    }
+
+    public function responseProvider(): iterable
+    {
+        $responses = [
+            ['status' => 200, 'content' => '<?xml version="1.0" encoding="utf-8" ?><acceptreport><statuscode>1</statuscode><statusmessage>Status code is not valid</statusmessage></acceptreport>', 'error_message' => 'Unable to send the SMS: "Status code is not valid".'],
+            ['status' => 500, 'content' => '', 'error_message' => 'Unable to send the SMS: "unknown error".'],
+        ];
+
+        foreach ($responses as $response) {
+            yield [$response['status'], $response['content'], $response['error_message']];
+        }
+    }
+
+    /**
+     * @dataProvider responseProvider
+     *
+     * @throws \Symfony\Component\Notifier\Exception\TransportExceptionInterface
```

Comment:
```suggestion
     *
     * @throws \Symfony\Component\Notifier\Exception\TransportExceptionInterface
```

can be removed in test